core/src/main/java/edu/wpi/grip/core/OperationDescription.java

```diff
@@ -0,0 +1,204 @@
+        return summary;
+    }
+
+    /**
+     * @return What category the operation falls under.  This is used to organize them in the GUI
+     */
+    public Category category() {
+        return category;
+    }
+
+    /**
+     * @return An {@link InputStream} of a 128x128 image to show the user as a representation of the operation.
+     */
+    public Optional<InputStream> icon() {
+        return Optional.ofNullable(icon);
+    }
+
+    /**
+     * @return Any old unique user-facing names of the operation. This is used to preserve compatibility with
+     * old versions of GRIP if the operation name changes.
+     */
+    public ImmutableSet<String> aliases() {
+        return aliases;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (this == o) return true;
+        if (!(o instanceof OperationDescription)) return false;
+
+        OperationDescription that = (OperationDescription) o;
+
+        if (name() != null ? !name().equals(that.name()) : that.name() != null) return false;
+        if (summary() != null ? !summary().equals(that.summary()) : that.summary() != null) return false;
+        if (category() != that.category()) return false;
+        if (icon() != null ? !icon().equals(that.icon()) : that.icon() != null) return false;
+        return aliases() != null ? aliases().equals(that.aliases()) : that.aliases() == null;
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = name() != null ? name().hashCode() : 0;
+        result = 31 * result + (summary() != null ? summary().hashCode() : 0);
+        result = 31 * result + (category() != null ? category().hashCode() : 0);
+        result = 31 * result + (icon() != null ? icon().hashCode() : 0);
+        result = 31 * result + (aliases() != null ? aliases().hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+                .add("name", name())
+                .add("summary", summary())
+                .add("aliases", aliases())
+                .add("category", category())
+                .toString();
+    }
+
+    /**
+     * Creates a new {@link Builder} instance to create a new {@code OperationDescription} object.
+     * <p>
+     * The created descriptor has a default category of {@link Category#MISCELLANEOUS MISCELLANEOUS} and no icon; use
+     * the {@link Builder#category(Category) .category()} and {@link Builder#icon(InputStream) .icon()} methods to
+     * override the default values.
+     */
+    public static Builder builder() {
+        return new Builder()
+                .category(Category.MISCELLANEOUS)
+                .icon(null);
+    }
+
+    /**
+     * Builder class for {@code OperationDescription}
+     */
+    public static final class Builder {
+        private String name;
+        private String summary = "PLEASE PROVIDE A DESCRIPTION TO THE OPERATION DESCRIPTION!";
```

Comment:
Because in tests its fine, when creating operations its good for simple testing.
Otherwise, it should show up in the pallette blatantly obvious that it is a problem.